tests/phpunit/CRM/Contact/Import/Parser/ContactTest.php

```diff
@@ -850,6 +850,45 @@ public function testImportAmbiguousStateCountry(): void {
     $this->assertEquals(array_search('United States', $countries), $addresses['values'][1]['country_id']);
   }
 
+  /**
+   * Test importing fields with various options.
+   *
+   * Ensure we can import multiple preferred_communication_methods, single
+   * gender, and single preferred language using both labels and values.
+   *
+   * @throws \CRM_Core_Exception @throws \CiviCRM_API3_Exception
+   */
+  public function testImportFieldsWithVariousOptions() {
+    $processor = new CRM_Import_ImportProcessor();
+    $processor->setContactType('Individual');
+    $processor->setMappingFields(
+      [
+        ['name' => 'first_name'],
+        ['name' => 'last_name'],
+        ['name' => 'preferred_communication_method'],
+        ['name' => 'gender_id'],
+        ['name' => 'preferred_language'],
+      ]
+    );
+    $importer = $processor->getImporterObject();
+    $fields = ['Ima', 'Texter', "SMS,Phone", "Female", "Danish"];
+    $importer->import(CRM_Import_Parser::DUPLICATE_NOCHECK, $fields);
+    $contact = $this->callAPISuccessGetSingle('Contact', ['last_name' => 'Texter']);
+
+    $this->assertEquals([4, 1], $contact['preferred_communication_method'], "Import multiple preferred communication methods using labels.");
+    $this->assertEquals(1, $contact['gender_id'], "Import gender with label.");
+    $this->assertEquals('da_DK', $contact['preferred_language'], "Import preferred language with label.");
+
+    $importer = $processor->getImporterObject();
+    $fields = ['Ima', 'Texter', "4,1", "1", "da_DK"];
```

Comment:
It's interesting that this works in the test, because I can't get numbers to work for this field in the UI, either before or after the patch or using single-value or multi.
Mystery?

Reply:
Huh - interesting. I put the values into the test because I thought, why not expand the test coverage? But didn't test entering the values in the UI. What happens when you try? Do you get an error?

Reply:
You can't get past the import screens to see what would happen - it gives you the excel file to download with errors which says it thinks the field has an invalid value.